src/adapter/src/catalog.rs

```diff
@@ -2485,6 +2485,40 @@ impl<S: Append> Catalog<S> {
         if let Some(system_parameter_frontend) = config.system_parameter_frontend {
             if !catalog.state.system_config().config_has_synced_once() {
                 tracing::info!("parameter sync on boot: start sync");
+
+                // We intentionally block initial startup, potentially forever,
+                // on initializing LaunchDarkly. This may seem scary, but the
+                // alternative is even scarier. Over time, we expect that the
+                // compiled-in default values for the system parameters will
```

Comment:
> we expect that the compiled-in default values for the system parameters will drift substantially from the defaults configured in LaunchDarkly

This does not need to be the case. We can periodically run a script that informs us if the LD default has deviated from the baked-in default.

Reply:
True! I updated the comment to mention that.